docs/databases/elasticsearch/a-guide-to-elasticsearch-plugins.md

```diff
@@ -0,0 +1,368 @@
+
+    POST /exampleindex/doc/1
+    {
+      "message": "this the value for the message field"
+    }
+
+There are a number of tools that can be used to issue this request. The simplest approach would be to use `curl` from the command line:
+
+    `curl -H'Content-Type: application/json' -XPOST localhost:9200/exampleindex/doc/1 -d '{ "message": "this the value for the message field" }'`
+
+Other alternatives include the [vim-rest-console](https://github.com/diepm/vim-rest-console), the Emacs plugin [es-mode](https://github.com/dakrone/es-mode), or the [Console](https://www.elastic.co/guide/en/kibana/current/console-kibana.html) plugin for Kibana. Use whichever tool is most convenient for you.
+
+### Prepare an Index
+
+Before installing any plugins, create a test index.
+
+1.  Create an index named `test` with one shard and no replicas:
+
+        POST /test
+        {
+          "settings": {
+            "index": {
+              "number_of_replicas": 0,
+              "number_of_shards": 1
+            }
+          }
+        }
+
+    {{< note >}}
+These settings are suitable for testing, but additional shards and replicas should be used in a production environment.
+{{< /note >}}
+
+2.  Add an example document to the index:
+
+        POST /test/doc/1
+        {
+          "message": "this is an example document"
+        }
+
+3.  Searches can be performed by using the `_search` URL endpoint. Search for "example" in the message field across all documents:
+
+        POST /_search
+        {
+          "query": {
+            "terms": {
+              "message": ["example"]
+            }
+          }
+        }
+
+    The Elasticsearch API should return the matching document.
+
+### Elasticsearch Attachment Plugin
+
+The attachment plugin lets Elasticsearch accept a base64-encoded document and index its contents for easy searching. This is useful for searching PDF or rich text documents with minimal overhead.
+
+1.  Install the `ingest-attachment` plugin using the `elasticsearch-plugin` tool:
+
+        sudo /usr/share/elasticsearch/bin/elasticsearch-plugin install ingest-attachment
+
+2.  Restart elasticsearch:
+
+        sudo systemctl restart elasticsearch
+
+3.  Confirm that the plugin is installed as expected by using the `_cat` API:
+
+        GET /_cat/plugins
+
+    The `ingest-attachment` plugin should be under the list of installed plugins.
+
+In order to use the attachment plugin, a _pipeline_ must be used to process base64-encoded data in the field of a document. An [ingest pipeline](https://www.elastic.co/guide/en/elasticsearch/reference/master/ingest.html) is a way of performing additional steps when indexing a document in Elasticsearch. While Elasticsearch comes pre-installed with some pipeline *processors* (which can perform actions such as removing or adding fields), the attachment plugin installs an additional processor that can be used when defining a pipeline.
```

Comment:
@Jskobos would you consider making these a new subheading? It's not necessary, but it might be a better visual break than the description paragraph and number restart